src/hooks/useStatusesFilter.ts

```diff
@@ -1,36 +1,67 @@
+import { useState } from "react";
 import { useLocation } from "react-router-dom";
 import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
 import { queryString } from "utils";
 
 const { parseQueryString } = queryString;
 /**
- * @param  {string} urlParam the param that will appear in the url search, e.g. `statuses`, `baseStatuses`
+ * @param {string} urlParam the param that will appear in the url search, e.g. `statuses`, `baseStatuses`
  * @param {boolean} page reset url page param to 0 if true
  * @param {boolean} resetPage update url page param to 0 if true
- * @return {[string[], (newValue: string[]) => void]}
- * pass first value in return array to `value` prop of dropdown component
- * pass second value in return array to `onChange` prop of dropdown component
+ * @return {UseStatusesFilterResult} Provides status filter input state and state management util functions
  */
 export const useStatusesFilter = (
   urlParam: string,
   resetPage?: boolean,
   sendAnalyticsEvent: (filterBy: string) => void = () => undefined
-): [string[], (newValue: string[]) => void] => {
+): UseStatusesFilterResult => {
   const { search } = useLocation();
   const updateQueryParams = useUpdateURLQueryParams();
+  const { [urlParam]: rawStatuses } = parseQueryString(search);
+  const urlValue = Array.isArray(rawStatuses)
+    ? rawStatuses
+    : [rawStatuses].filter((v) => v);
+
+  const [inputValue, setInputValue] = useState(urlValue);
```

Comment:
This stores the state both internally and on the url is this necessary? It seems like it would add complexity to keep these 2 values in sync

Reply:
the useState acts as a buffer for changes that will be committed to the url after the sure submits enter. the input view state and url state can diverge between the time the user types and submits enter.